libs/map/gps_track.cpp

```diff
@@ -126,18 +96,24 @@ void GpsTrack::ScheduleTask()
 {
   lock_guard<mutex> lg(m_threadGuard);
 
+  /// @todo Replace with !m_thread.joinable() ?
   if (m_thread.get_id() == std::thread::id())
```

Comment:
Joinable [returns true](https://godbolt.org/z/jf4jYhEv9) when thread has finished execution but wasn't joined yet.

What was the idea here? `m_thread.get_id()` is either not initialized, or initialized and returns _some_ other thread id. If `SheduleTask` is called from main, then every time a new thread is created, right? Why?